packages/elements/src/react/common/form/otp.tsx

```diff
@@ -88,173 +87,70 @@ const OTPInputStandard = React.forwardRef<HTMLInputElement, Omit<OTPInputProps,
  */
 const OTPInputSegmented = React.forwardRef<HTMLInputElement, Required<Pick<OTPInputProps, 'render'>> & OTPInputProps>(
   function OTPInput(props, ref) {
-    const {
-      className: userProvidedClassName,
-      render,
-      length = OTP_LENGTH_DEFAULT,
-      autoSubmit = false,
-      ...rest
-    } = props;
-
-    const value = props.value as string;
-    const previousValue = usePrevious(value);
+    const { className, render, length = OTP_LENGTH_DEFAULT, autoSubmit = false, ...rest } = props;
 
     const innerRef = React.useRef<HTMLInputElement>(null);
-    // This ensures we can access innerRef internally while still exposing it via the ref prop
-    React.useImperativeHandle(ref, () => innerRef.current as HTMLInputElement, []);
-
     const [selectionRange, setSelectionRange] = React.useState<SelectionRange>(props.autoFocus ? ZERO : OUTSIDE);
-    const isFocused = useFocus(innerRef);
-
-    // Attach this component's UI state "selectionRange" to the native .select() to update it on focus changes
-    React.useEffect(() => {
-      const element = innerRef.current;
-
-      if (!element) return;
-
-      const originalSelect = element.select.bind(element);
-      element.select = () => {
-        originalSelect();
-        setSelectionRange([0, element.value.length]);
-      };
-    }, []);
```

Comment:
Why did we use the native `.select()` here?

Reply:
Might have been a left-over from testing, I think it should be covered by the existing listeners